src/scaladoc/scala/tools/nsc/doc/base/CommentFactoryBase.scala

```diff
@@ -544,6 +562,376 @@ trait CommentFactoryBase { this: MemberLookupBase =>
       HorizontalRule()
     }
 
+    /** {{{
+      * table      ::= '{|' '\n' { caption } { header } { rows } '\n' '|}'
+      * caption    ::= '|+' content
+      * header     ::= headerCell+ { nBang }
+      * headerCell ::= nBang content
+      * nBang      ::= '!'+
+      * rows       ::= row+
+      * row        ::= cell+ nPipe
+      * cell       ::= nPipe content
+      * nPipe      ::= '|'+
+      * content    ::= { '\n'+ } { inline } { content }
+      * }}}
+      */
+    def table(): Block = {
+
+      /* Helpers */
+
+      def peek(tag: String): Unit = {
+        val peek: String = buffer.substring(offset)
+        val limit = 60
+        val limitedPeek = peek.substring(0, limit min peek.length)
+        println(s"peek: $tag: '$limitedPeek'")
+      }
+
+      def nextIsCaptionStart = check(TableCaptionStart)
+
+      def nextIsExclusivelyCellStart = {
+        /* Disambiguate cell markdown from other table markdown sequences */
+        val notCellStart = TableAllMarks filterNot (_ == TableCellStart)
+
+        check(TableCellStart) && !notCellStart.exists(check)
+      }
+
+      /* No other table mark starts with ! so the test is whitelist only */
+      def nextIsExclusivelyHeaderCellStart = check(TableHeaderCellStart)
+
+      val outerMarks = Set(TableStart, TableEnd)
+
+      val innerMarks = TableAllMarks filterNot outerMarks
+
+      def nextIsInnerMark = innerMarks.exists(check) && !outerMarks.exists(check)
+
+      /* Accumulated state */
+
+      var caption: Option[Caption] = None
+
+      var header: Option[Row] = None
+
+      val rows = mutable.ListBuffer.empty[Row]
+
+      val cells = mutable.ListBuffer.empty[Cell]
+
+      def finalizeCells(): Unit = {
+        if (cells.nonEmpty) {
+          rows += Row(cells.toList)
+        }
+        cells.clear()
+      }
+
+      def finalizeHeaderCells(): Unit = {
+        if (cells.nonEmpty) {
+          if (header.isDefined) {
+            reportError(pos, "more than one table header")
+          } else {
+            header = Some(Row(cells.toList))
+          }
+        }
+        cells.clear()
+      }
+
+      def checkAny(terminators: List[String]) = terminators.exists(check)
+
+      def endOfInlineWhenTableMarkOnNextLine = checkAny(TableAllMarksOnNextLine)
+
+      def isEndOfText = char == endOfText
+
+      def isNewline = char == endOfLine
+
+      def isNewlineAnyTableMarks = checkAny(TableAllMarksOnNextLine)
```

Comment:
`isNewlineAnyTableMarks` is identical to `endOfInlineWhenTableMarkOnNextLine`. There can only be one `table-mark-on-next-line` incumbent!!!!